tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/SdkCodeTool.cs

```diff
@@ -0,0 +1,467 @@
+                }
+
+                logger.LogInformation($"Repository root path: {sdkRepoRoot}");
+
+                // Get the build script path and resolve full path
+                string fullBuildScriptPath;
+                try
+                {
+                    var buildScriptPath = await GetScriptPathFromConfigAsync(sdkRepoRoot, "packageOptions/buildScript/path");
+                    logger.LogInformation($"Build script path: {buildScriptPath}");
+
+                    // Resolve the full path of the build script
+                    fullBuildScriptPath = Path.IsPathRooted(buildScriptPath) 
+                        ? buildScriptPath 
+                        : Path.Combine(sdkRepoRoot, buildScriptPath);
+
+                    if (!File.Exists(fullBuildScriptPath))
+                    {
+                        return CreateFailureResponse($"Build script not found at: {fullBuildScriptPath}");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    return CreateFailureResponse($"Failed to get build script path: {ex.Message}");
+                }
+
+                // Run the build script
+                logger.LogInformation($"Executing build script: {fullBuildScriptPath}");
+
+                // TODO: change --module-dir to --project-path
+                var options = new ProcessOptions(
+                    fullBuildScriptPath,
+                    new[] { "--module-dir", projectPath },
+                    logOutputStream: true,
+                    workingDirectory: sdkRepoRoot,
+                    timeout: TimeSpan.FromMilliseconds(commandTimeout)
+                );
+                var buildResult = await processHelper.Run(options, ct);
+                var trimmedBuildResult = (buildResult.Output ?? string.Empty).Trim();
+                if (buildResult.ExitCode != 0)
+                {
+                    return CreateFailureResponse($"Build script failed with exit code {buildResult.ExitCode}. Output:\n{trimmedBuildResult}");
+                }
+
+                logger.LogInformation("Build script execution completed");
+                return CreateSuccessResponse($"Build completed successfully. Output:\n{trimmedBuildResult}");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error occurred while building SDK");
+                return CreateFailureResponse($"An error occurred: {ex.Message}");
+            }
+        }
+
+        // Run language-specific script to generate the SDK code from 'tspconfig.yaml'
+        private async Task<DefaultCommandResponse> GenerateSdkFromTspConfigAsync(string localSdkRepoPath, string tspConfigPath, string specCommitSha, string specRepoFullName, string emitterOptions, CancellationToken ct)
+        {
+            // white spaces will be added by agent when it's a URL
+            tspConfigPath = tspConfigPath.Trim();
+            
+            // Validate inputs
```

Comment:
tsp-client already does this validation, no? Do you think we could just skip this validation and let tsp-client handle it?

Reply:
I agree with it to avoid the duplicate the validation. I filed [an issue](https://github.com/Azure/azure-sdk-tools/issues/11900) for 'tsp-client'. I'll remove this validation once tsp-client enhances it.